### PR TITLE
STM32: Fix LSE hang at startup

### PR DESCRIPTION
**Pull request by hierophect**
This PR attempts to fix the low power issue in #2861 by adding a timeout for the LSE that attempts to swap over to the LSI on failure. Drafted because it still does not fully resolve the problem.

### CODE REVIEW

File: ports/stm/supervisor/port.c [hunk 1]
```diff
@@ -172,22 +172,36 @@ safe_mode_t port_init(void) {
...
-    while(__HAL_RCC_GET_FLAG(RCC_FLAG_LSERDY) == RESET) {}
+    bool lse_setupsuccess = true;
+    uint32_t i = 0;
+    while(__HAL_RCC_GET_FLAG(RCC_FLAG_LSERDY) == RESET) {
+        if ( ++i > 1000000 )
+        {
+            lse_setupsuccess = false;
+            __HAL_RCC_LSE_CONFIG(RCC_LSE_OFF);
+            __HAL_RCC_LSI_ENABLE();
+            rtc_clock_frequency = LSI_VALUE;
```
Comment (tannewt) [inline ports/stm/supervisor/port.c:183]: break here to make it clear the loop is done.

**Comment by jerryneedell** [comment]
FYi -- i tried this PR on a feather stm32f405 and it still does not boot.

**Comment by hierophect** [comment]
Ok, fixes are in @tannewt. @jerryneedell, the stack sizes have been adjusted - if you could run your stack code on this that would be very helpful. 

Important note! Once programmed you may need to reset the power before these fixes will take effect. It appears settings in the Backup Domain like to persist past writes and such, and I've observed that sometimes the chip will not boot when switching between the LSI and LSE, as we saw briefly on the call today. A simple reset does not do the trick - you must power cycle.

**Comment by hierophect** [comment]
Also, since the H7 and F7 have different settings for the Backup Domain, and I don't think we've effectively tested them, I've disabled LSE on them entirely until I have a chance to revisit this and hopefully move all this RCC setup code into Peripherals management where it probably belongs along with the rest of the oscillator stuff.

**Comment by jerryneedell** [comment]
sorry I'm late, but it works great on my feather_stm32405 -- also resolves the I2C and SPI issues.

**Pull request merged**